### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by codesalley**
bubble sort implementation. iterate through the input and check if the current value is greater than the adjacent value. if yes swap else move to the next element.

### CODE REVIEW

File: bubble_sort_by.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    arr.each_with_index do |_x, i|
+      left = arr[i]
+      arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[-1] != left && yield(arr[i], arr[i + 1]).positive?
+    end
+  end
+  arr
+end
+p bubble_sort_by(%w[hi hello hey]) { |left, right|
+  left.length - right.length
+}
```
Comment (BrianSammit) [inline bubble_sort_by.rb]: Lines 10, 11 and 12. where you have 
'p bubble_sort_by(%w[hi hello hey]) { |left, right|
  left.length - right.length
}'
 it is a debugging code. Kindly deleted or move to another branch. :wrench: 

debugging_requirement

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  input_arr.length.times do
+    input_arr.each_with_index do |num, index|
+      next unless input_arr[index + 1]
+
+      input_arr[index], input_arr[index + 1] = input_arr[index + 1], input_arr[index] if num > input_arr[index + 1]
+    end
+  end
+  input_arr
+end
+puts bubble_sort([4, 3, 78, 2, 0, 2])
```
Comment (BrianSammit) [inline bubble_sort.rb]: Line 11. where you have 'puts bubble_sort([4, 3, 78, 2, 0, 2])' it is a debugging code. Kindly deleted or move to another branch. :wrench: 

debugging_requirement

**Review by BrianSammit — CHANGES_REQUESTED** [review]
REQUIRED CHANGES!! :recycle: 
-----------
Hello team!! :wave: you have an excellent project :100: :1st_place_medal: 
There are a few changes to make. see them below :point_down: 
-----------

- [ ] In your readme, you have a  misspelling, You write 'repeates' and it should be 'repeats'.  Kindly fix it so your Readme gonna looks more professional. Before you submit for review check the grammar with Grammarly to ensure you don't have any grammar errors on your project.
- [ ] In your repo you have a directory called '.idea'. remember that files not related to the project should not be committed and added to .gitignore.
- [ ] You have a debugging code, remember that debugging code should not be committed. You can keep your personal test on a different branch if you want to ever revisit them. [Here](https://questions.microverse.org/t/debugging-instructions-code/346) you can read the discussion about it.
----------
Make these minor changes and send it for another review :muscle: 
HAPPY CODING!!! :smile: 
MB5W4D

**Comment by MarcusAl** [comment]
Fixed the errors. And the debugging code.

**Review by jubaan — APPROVED** [review]
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> 
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🥳 🎊 🎉 🥳  APPROVED 🥳 🎊 🎉 🥳 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉
</h1>
<h2 align="center">
  What’s done is done. What’s gone is gone. <br />
  One of life’s lessons is always moving on. <br />
  It’s okay to look back to see how far you’ve come but <br /> 
  KEEP MOVING FORWARD.
</h2>

<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>


<h3 id="optional"> 1. 💡 Nice to have and other comments </h3>
<details close="close">
<summary> [click to open] If you have something here, even when approved, it's just a little push to improve your project.</h3></summary>


- [ ] Nothing more to add.


</details>

<h3 id="contact"> 2. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    Julio Añoveros
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 3. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<p> How does it feel? Is nice to advance in the program, right? It keeps you motivated even with all the struggles and takes you one step closer to the goal 🎯!. I wan't you to keep advancing so if you feel stuck and/or need help, remember there is always someone you can reach to keep moving forward, doesn't mater how "dum" you think the questions is.</p>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> ... it would be great to own one 😄. </p>

<h1 align="center"> See you a the top 🚀!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

**Pull request merged**